### PR TITLE
Freemius integration

### PR DESCRIPTION
**Pull request by sc0ttkclark**
## Description
Integrates Freemius

## ChangeLog
* Added: Freemius integration to support our new Friends of Pods add-ons and enable us to let people opt-in to usage stats for planning future compatibility decisions.

## Checklist:
- [x] My code is tested.
- [x] My code follows the WordPress code style.
- [x] My code follows has proper inline documentation.

### CODE REVIEW

File: classes/PodsInit.php [hunk 1]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+	 */
+	public function freemius() {
+		if ( $this->freemius ) {
+			return $this->freemius;
+		}
+
+		require_once dirname( __DIR__ ) . '/vendor/freemius/wordpress-sdk/start.php';
+
+		try {
+			$this->freemius = fs_dynamic_init( [
```
Comment (tr1b0t) [inline classes/PodsInit.php]: * :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line
* :no_entry_sign: Short array syntax is not allowed

<sub>hash: d9bbdd1246b4262b0b6007c8bd6c6c7e</sub>

File: classes/PodsInit.php [hunk 2]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+				'id'             => '5347',
+				'slug'           => 'pods',
+				'type'           => 'plugin',
+				'public_key'     => 'pk_737105490825babae220297e18920',
+				'is_premium'     => false,
+				'has_addons'     => true,
+				'has_paid_plans' => false,
+				'menu'           => [
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: bac338386e05344c9b61cd11131ba607</sub>

File: classes/PodsInit.php [hunk 3]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+					'slug'        => 'pods-settings',
+					'contact'     => false,
+					'support'     => false,
+					'affiliation' => false,
+					'account'     => true,
+					'pricing'     => false,
+					'addons'      => true,
+					'parent'      => [
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: b8df361b036f4c0160bfbab8361509da</sub>

File: classes/PodsInit.php [hunk 4]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+						'slug' => 'pods',
+					],
+				],
+			] );
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: eb8a502c2f7e111982b78fe965bcf4ad</sub>

File: classes/PodsInit.php [hunk 5]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+
+			$this->override_freemius_strings();
+
+			add_filter( 'fs_plugins_api', [ $this, 'filter_freemius_plugins_api_data' ], 15 );
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: c1865f370a0d47d8fec6f359e997649d</sub>

File: classes/PodsInit.php [hunk 6]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+
+			$this->freemius->add_filter( 'templates/add-ons.php', [ $this, 'filter_freemius_addons_html' ] );
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 36fabc184aef949386914b37e23f4330</sub>

File: classes/PodsInit.php [hunk 7]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+			$this->freemius->add_filter( 'download_latest_url', [ $this, 'get_freemius_action_link' ]  );
```
Comment (tr1b0t) [inline classes/PodsInit.php]: * :no_entry_sign: Expected 1 spaces before closing bracket; 2 found
* :no_entry_sign: Short array syntax is not allowed

<sub>hash: b4fdd4035cfafa814adb9e5565de0d1d</sub>

File: classes/PodsInit.php [hunk 8]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+		}
+
+		return $this->freemius;
+	}
+
+	/**
+	 * Override Freemius strings.
+	 */
+	public function override_freemius_strings() {
+		$override_text = [
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 01e13b438f9b1fb69bc5de6e3fd4cfb2</sub>

File: classes/PodsInit.php [hunk 9]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+
+	/**
+	 * Get list of add-ons for Freemius.
+	 *
+	 * @since TBD
+	 *
+	 * @return array List of add-ons for Freemius.
+	 */
+	public function get_freemius_addons() {
+		return [
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 91011d2f9087ba6a4fbfb7710aa9df62</sub>

File: classes/PodsInit.php [hunk 10]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+
+	/**
+	 * Get list of Friends-only add-ons for Freemius.
+	 *
+	 * @since TBD
+	 *
+	 * @return array List of Friends-only add-ons for Freemius.
+	 */
+	public function get_freemius_friends_addons() {
+		return [
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Short array syntax is not allowed

<sub>hash: 70c427abf215be0af89744a94cd66934</sub>

File: readme.txt [hunk 11]
```diff
@@ -4,7 +4,7 @@ Donate link: https://pods.io/friends-of-pods/
 Tags: pods, custom post types, custom taxonomies, content types, custom fields, cck, database, user fields, comment fields, media fields, relationships, drupal
 Requires at least: 4.5
 Tested up to: 5.4
-Requires PHP: 5.3
+Requires PHP: 5.4
```
Comment (sc0ttkclark) [inline readme.txt]:
```suggestion
Requires PHP: 5.3
```

File: classes/PodsInit.php [hunk 12]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+	public function freemius() {
+		if ( $this->freemius ) {
+			return $this->freemius;
+		}
+
+		require_once dirname( __DIR__ ) . '/vendor/freemius/wordpress-sdk/start.php';
+
+		try {
+			$this->freemius = fs_dynamic_init( array(
+				'id'             => '5347',
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Opening parenthesis of a multi-line function call must be the last content on the line

<sub>hash: 77644cb87af2638f313c6bde899d9cf3</sub>

File: classes/PodsInit.php [hunk 13]
```diff
@@ -150,6 +159,176 @@ public function plugins_loaded() {
...
+					'affiliation' => false,
+					'account'     => true,
+					'pricing'     => false,
+					'addons'      => true,
+					'parent'      => array(
+						'slug' => 'pods',
+					),
+				),
+			) );
+
```
Comment (tr1b0t) [inline classes/PodsInit.php]: :no_entry_sign: Closing parenthesis of a multi-line function call must be on a line by itself

<sub>hash: ff8fc95f402a72000c63400456631690</sub>

File: ui/admin/settings.php [hunk 14]
```diff
@@ -7,11 +7,11 @@
 			<?php
 			$default = 'tools';
 
-			$tabs = [
+			$tabs = array(
```
Comment (tr1b0t) [inline ui/admin/settings.php]: :no_entry_sign: Overriding WordPress globals is prohibited. Found assignment to $tabs

<sub>hash: c21becd954a641d2a0b5eae2b4d99ec3</sub>

**Pull request merged**